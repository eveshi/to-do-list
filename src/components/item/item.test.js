import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Item from './item';

configure({ adapter: new Adapter() });

describe('<Item />', () => {
  // beforeAll(() => {
  //   const wrapper = shallow(<Item />);
  // });

  it('click button to show text', () => {
    const wrapper = mount(<Item />);
    expect(wrapper.find('p')).toHaveLength(1);
    const instance = wrapper.instance();

    instance.clickHandler();
    wrapper.update();
    expect(wrapper.find('p')).toHaveLength(2);
  });
});

describe('snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Item />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
