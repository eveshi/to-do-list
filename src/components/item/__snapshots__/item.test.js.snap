// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly 1`] = `
<div>
  <p>
    there is music
  </p>
  <button
    onClick={[Function]}
  >
    change
  </button>
</div>
`;
